pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -121,18 +129,41 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1.ApiServerSour
 	return nil
 }
 
-func (r *Reconciler) createReceiveAdapter(ctx context.Context, src *v1.ApiServerSource, sinkURI string) (*appsv1.Deployment, error) {
+func (r *Reconciler) namespacesFromSelector(ctx context.Context, src *v1.ApiServerSource) ([]string, error) {
+	if src.Spec.NamespaceSelector == nil {
+		return []string{src.Namespace}, nil
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(src.Spec.NamespaceSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	namespaces, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We should use the lister instead of the client to avoid API server overload and also because we have namespaces in the cache